### ISSUE TITLE
username already exists

### ISSUE THREAD

**Issue by test2a** [Chocobozzz/PeerTube#3093]
**What happened?**
https://videos.lukesmith.xyz
I tried to register for this website and it asks to set a username and channel name. After i press done, it says "User with this username or email already exists."

The software should check at the time of entering the username if the same already exists instead of finishing the form first, same for channel name and email address. Also, why is password asked only once? isnt it standard practice to make the user enter password twice?

![Screenshot_20200816_204841](https://user-images.githubusercontent.com/39449563/90337726-0237f700-e002-11ea-8dd9-104b8ca1422a.png)
![Screenshot_20200816_204908](https://user-images.githubusercontent.com/39449563/90337728-0401ba80-e002-11ea-9b37-3734f4b1bbc4.png)
![Screenshot_20200816_204921](https://user-images.githubusercontent.com/39449563/90337730-0532e780-e002-11ea-8ca8-e27e6543fbd9.png)

**What do you expect to happen instead?**
validation should happen when entering the username/email/channel name right next to the box instead of finishing the form and then informing the user.  This way the user can change their details until the software says okay

**Additional information**
* PeerTube version or URL:2.3.0
* Browser name/version:
* NodeJS version:

* Link to browser console log if useful:
* Link to server log if useful (journalctl or /var/www/peertube/storage/logs/):

**Comment by rigelk** [comment]
> Also, why is password asked only once? isnt it standard practice to make the user enter password twice?

Not really. Confirmation fields were largely introduced due to password masking, which we can just solve with a button allowing to show the password entered. Some big names (i.e. Facebook) also don't use confirmation fields ; I guess that has more to do with how little friction to register they want, but still, it means that confirmation fields are not necessary *per se*.

**Comment by test2a** [comment]
@rigelk that sounds fair enough. a mask/unmask button in the password field would be a good compromise. In my defense, i have been mostly seeing the two password boxes in signup for over 15 years so thats why i said that.
